README.md

```diff
@@ -75,6 +75,8 @@ Before extending into new frontiers, we need to improve the loaders API enough t
 - [ ] Hooks for customizing the REPL, including transpilation and tab completion. Support users pasting TypeScript (or CoffeeScript or whatever) into the REPL and having just as good an experience as with plain JavaScript.
 
    - [ ] Support top-level `await` in the REPL API, if possible.
+     
+   - [ ] Add support for `--eval` (or `--print`) CLI flags, extending over to support eval [Worker](https://nodejs.org/api/worker_threads.html) as well.
```

Comment:
```suggestion
- [ ] Add support `--eval` CLI flag (as well as `Worker` eval if that ever becomes a thing).
```
When using `--print`, `--eval` is implied.
I'd say we could skip the `Worker` mention here, I think it goes without saying that if support was added, it would work similarly as `--input-type=module --eval`, but I guess it doesn't matter to have it anyway.

Reply:
> When using --print, --eval is implied.

Is it? There's a pretty big incompatibility that `--eval` supports `--input-type=module` and `--print` doesn't.

If we're both indifferent about the Worker mention, I'll keep it as per @GeoffreyBooth's request

Reply:
> > When using --print, --eval is implied.
> 
> Is it?

Yes, doing `node --print whatever` or `node --print --eval whatever` is equivalent, and go through the same code path.



> There's a pretty big incompatibility that `--eval` supports `--input-type=module` and `--print` doesn't.

You cannot print ESM because it's async is a thing, but that's not related to loaders, it has nothing to do with how we design the new hook. No matter whether support for that was somehow added later, it wouldn't change the design of the hook API.

Reply:
I see, thanks for explaining.

FWIW @GeoffreyBooth didn't seem to think it would be feasible with `--print` in https://github.com/nodejs/loaders/issues/116#issuecomment-1816817724. As long as people are drawing difference inferences, I think there's value in being explicit here.

Do you still insist I remove `--print`?

Reply:
Here’s what I understand from this line: we’d need a hook that takes a string and returns another string to be `eval`d, either to hook into `--eval`, `Worker` constructor, or stdin (`echo '1+1' | node`). 

I think specifying `--print` and `Worker` only muddies the water, as it’s listing some specific use cases of a more global thing. 

Discussing `--print` in particular is really off-topic because:
- it implies `--eval`, so we couldn’t support one without the other. Listing them both will likely confuse the reader as if it was two separated input types. 
- the fact that it supports or not ESM does not change anything to the design of the API. If it supports only CJS, so be it, I don’t understand why that would be a concern to design that new hook.

Reply:
Given my current understanding of the Loader/Hook API is that it only handles ESM, I would personally be skeptical if it would work with `--print` (or `--eval` with `--input-type=commonjs`) because they only allow CommonJS. And being specific would help.

Anyway, since you insist, I'll remove the `--print`. And since you don't actually seem to be indifferent about having `Worker` either, I'll remove that too.

Reply:
I think there’s a misunderstanding here: the Hooks API is not supposed to be limited to ESM. I don’t think your skepticism is unfounded, but I think it shouldn’t block us from supporting it in the Hooks API.

Reply:
Yeah, I also thought so too based on https://github.com/nodejs/node/issues/50649#issuecomment-1818975732

Do you know if the Hook API supports loading CJS now? Or you're saying that's on the roadmap and will be supported in the future? AFAIK I believe it doesn't right now.

Reply:
> `Worker` eval if that ever becomes a thing

What does this mean? Worker eval is supported already.

Reply:
> Do you know if the Hooks API supports loading CJS now?

It does, it was added by https://github.com/nodejs/node/pull/47999

Reply:
Gotcha! That's great. Thank you for sharing.